examples/dcgan/dcgan.gop

```diff
@@ -80,4 +177,84 @@ func main() {
 	}
 
 	dataloader := data.NewLoader(dataset, *batchSize, true /*shuffle*/, *workers)
+
+	device := torch.Device("cpu")
+	if *cuda {
+		device = torch.Device("cuda:0")
+	}
+
+	netG := NewGenerator(*npgu, *nz, *ngf, nc)
+	netG.To(device)
+	netG.Apply(weightInit)
+	if *pathG != "" {
+		netG.LoadStateDict(torch.Load(*pathG))
+	}
+
+	netD := NewDiscriminator(*ngpu, *ndf, *nc)
+	netD.To(device)
+	netD.Apply(weightInit)
+	if *pathD != "" {
+		netG.LoadStateDict(torch.Load(*pathD))
+	}
+
+	criterion := torch.BCELoss()
+	fixedNoise := torch.RandN(*batchSize, *nz, 1, 1, device)
+	realLabel := 1
+	fakeLabel := 0
+
+	optimizerD := torch.Adam(netD.Parameters(), *lr, []float64{*beta1, 0.999})
+	optimizerG := torch.Adam(netG.Parameters(), *lr, []float64{*beta1, 0.999})
+
+	if *dryRun {
+		*niter = 1
+	}
+
+	for epoch := 0; epoch < *niter; epoch++ {
+		for i, data := range dataloader {
```

Comment:
This is an illegal Go syntax. Let us use valid Go syntax in this .gop file, so we can finally get it running in a few days.

Reply:
@Yancey1989 Do you have some suggestions on this? Thansk!

Reply:
Done. Follow the design of the scanner package of Go.

```
for i, data := 0, dataloader.ReadBatch(); data != dataloader.EOF; i, data = i + 1, dataloader.ReadBatch() {
}
```

Reply:
Maybe we don't need need `for i ...`, which is C++ pattern, what do you think about the following code:

``` golang
loader = torch.NewLoader(dataset, 2)
for loader.Scan() {
    batch := loader.Batch()
    fmt.Println(batch.Data)
    fmt.Println(batch.Target)
}
```

You can find more details on issue #30